src/mainboard/protectli/vault_adl_p/gpio.c

```diff
@@ -0,0 +1,631 @@
+	/* GPP_A12 - GPIO */
+	PAD_NC(GPP_A12, NONE),
+	/* GPP_A13 - GPIO */
+	PAD_NC(GPP_A13, NONE),
+	/* GPP_A14 - DDSP_HPD3 */
+	PAD_CFG_NF(GPP_A14, NONE, DEEP, NF2),
+	/* GPP_A15 - GPIO */
+	PAD_NC(GPP_A15, NONE),
+	/* GPP_A16 - GPIO */
+	PAD_NC(GPP_A16, NONE),
+	/* GPP_A17 - GPIO */
+	PAD_NC(GPP_A17, NONE),
+	/* GPP_A18 - DDSP_HPDB */
+	PAD_CFG_NF(GPP_A18, NONE, DEEP, NF1),
+	/* GPP_A19 - GPIO */
+	PAD_NC(GPP_A19, NONE),
+	/* GPP_A20 - GPIO */
+	PAD_NC(GPP_A20, NONE),
+	/*
+	 * Below pins are configured before FSP-S multiphase init,
+	 * see typec_aux_bias_pads in devicetree.cb
+	 */
+	/* GPP_A21 - USB_C_GPP_A21 */
+	/* PAD_CFG_NF(GPP_A21, NONE, PLTRST, NF6), */
+	/* GPP_A22 - USB_C_GPP_A22 */
+	/* PAD_CFG_NF(GPP_A22, NONE, PLTRST, NF6), */
+	/* GPP_A23 - ESPI_CS1# */
+	PAD_CFG_NF(GPP_A23, UP_20K, DEEP, NF1),
+	/* GPP_ESPI_CLK_LOOPBK - GPP_ESPI_CLK_LOOPBK */
+	PAD_CFG_NF(GPP_ESPI_CLK_LOOPBK, NONE, DEEP, NF1),
+
+	/* ------- GPIO Community 1 ------- */
+
+	/* ------- GPIO Group GPP_S ------- */
+
+	/* GPP_S0 - GPIO */
+	PAD_NC(GPP_S0, NONE),
+	/* GPP_S1 - GPIO */
+	PAD_NC(GPP_S1, NONE),
+	/* GPP_S2 - GPIO */
+	PAD_NC(GPP_S2, NONE),
+	/* GPP_S3 - GPIO */
+	PAD_NC(GPP_S3, NONE),
+	/* GPP_S4 - GPIO */
+	PAD_NC(GPP_S4, NONE),
+	/* GPP_S5 - GPIO */
+	PAD_NC(GPP_S5, NONE),
+	/* GPP_S6 - GPIO */
+	PAD_NC(GPP_S6, NONE),
+	/* GPP_S7 - GPIO */
+	PAD_NC(GPP_S7, NONE),
+
+	/* ------- GPIO Group GPP_H ------- */
+
+	/* GPP_H0 - GPIO */
+	PAD_NC(GPP_H0, NONE),
+	/* GPP_H1 - GPIO */
+	PAD_NC(GPP_H1, NONE),
+	/* GPP_H2 - GPIO */
+	PAD_NC(GPP_H2, NONE),
+	/* GPP_H3 - GPIO */
+	PAD_NC(GPP_H3, NONE),
+	/* GPP_H4 - GPIO */
+	PAD_NC(GPP_H4, NONE),
+	/* GPP_H5 - GPIO */
+	PAD_NC(GPP_H5, NONE),
+	/* GPP_H6 - I2C1_SDA */
+	PAD_CFG_NF(GPP_H6, NONE, DEEP, NF1),
+	/* GPP_H7 - I2C1_SCL */
+	PAD_CFG_NF(GPP_H7, NONE, DEEP, NF1),
+	/* GPP_H8 - GPIO */
+	PAD_NC(GPP_H8, NONE),
+	/* GPP_H9 - GPIO */
+	PAD_NC(GPP_H9, NONE),
+	/* GPP_H10 - GPIO */
+	PAD_NC(GPP_H10, NONE),
+	/* GPP_H11 - GPIO */
+	PAD_NC(GPP_H11, NONE),
+	/* GPP_H12 - GPIO */
+	PAD_NC(GPP_H12, NONE),
+	/* GPP_H13 - GPIO */
+	PAD_NC(GPP_H13, NONE),
+	/* GPP_H14 - GPIO */
+	PAD_NC(GPP_H14, NONE),
+	/* GPP_H15 - DDPB_CTRLCLK */
+	PAD_CFG_NF(GPP_H15, NONE, DEEP, NF1),
+	/* GPP_H16 - GPIO */
+	PAD_NC(GPP_H16, NONE),
+	/* GPP_H17 - DDPB_CTRLDATA */
+	PAD_CFG_NF(GPP_H17, NONE, DEEP, NF1),
+	/* GPP_H18 - PROC_C10_GATE# */
+	PAD_CFG_NF(GPP_H18, NONE, DEEP, NF1),
+	/* GPP_H19 - SRCCLKREQ4# */
+	PAD_NC(GPP_H19, NONE),
+	/* GPP_H20 - GPIO */
+	PAD_NC(GPP_H20, NONE),
+	/* GPP_H21 - GPIO */
+	PAD_NC(GPP_H21, NONE),
+	/* GPP_H22 - GPIO */
+	PAD_NC(GPP_H22, NONE),
+	/* GPP_H23 - GPIO */
+	PAD_NC(GPP_H23, NONE),
+
+	/* ------- GPIO Group GPP_D ------- */
+
+	/* GPP_D0 - GPIO */
+	PAD_NC(GPP_D0, NONE),
+	/* GPP_D1 - GPIO */
+	PAD_NC(GPP_D1, NONE),
+	/* GPP_D2 - GPIO */
+	PAD_NC(GPP_D2, NONE),
+	/* GPP_D3 - GPIO */
+	PAD_NC(GPP_D3, NONE),
+	/* GPP_D4 - GPIO */
+	PAD_NC(GPP_D4, NONE),
+	/* PAD_NC - SRCCLKREQ0# */
+	PAD_NC(GPP_D5, NONE),
```

Comment:
Isn't it routed to I226?

Reply:
Two out of 4 i226s have CLKREQ signal connected. This might also affect devicetree.
Even if it should stay this way, I would be interested to learn why it should stay this way.

Reply:
Yes, but it doesn't make much sense if just 2 out of all devices have this CLKREQ enabled. The power savings from CLOCK PM and ASPM L1 will be marginal... So to save time and avoid problems I left ti as disabled. coreboot only has a global switch for CLOCK PM and ASPM L1, so it may break other devices for this board.

Reply:
IMHO, it would be great to document such decisions as a comment

Reply:
And for whom should I give that comment?  Only a handful of people actually read the code.

Besides, putting so much comments would probably cause unnecessary discussions on upstream review.

Reply:
Added comments aroudn CLKREQs

Reply:
Thank you.